frontend/src/metabase/query_builder/containers/QueryBuilder.jsx

```diff
@@ -354,19 +364,62 @@ function QueryBuilder(props) {
     }
   });
 
+  const { isRunning } = uiControls;
+
+  const [sendNotification, setSendNotification] = useState(false);
+  const [showToaster, setShowToaster] = useState(false);
+
+  const toastTrigger = useLoadingTimer(isRunning, 5000);
+  const [requestPermission, showNotification] = useWebNotification();
+
+  useEffect(() => {
+    if (toastTrigger) {
+      setShowToaster(true);
+    }
+  }, [toastTrigger]);
+
+  useEffect(() => {
+    if (!isRunning) {
+      setShowToaster(false);
+    }
+    if (!isRunning && sendNotification) {
+      showNotification(
+        `All Set! You question is ready.`,
+        `${card.name} is loaded.`,
+      );
+    }
+  }, [isRunning, sendNotification, showNotification, card.name]);
+
+  const handleToastConfirm = async () => {
+    const result = await requestPermission();
+    if (result === "granted") {
+      setShowToaster(false);
+      setSendNotification(true);
+    }
+  };
+
   return (
-    <View
-      {...props}
-      modal={uiControls.modal}
-      recentlySaved={uiControls.recentlySaved}
-      onOpenModal={openModal}
-      onCloseModal={closeModal}
-      onSetRecentlySaved={setRecentlySaved}
-      onSave={handleSave}
-      onCreate={handleCreate}
-      handleResize={forceUpdateDebounced}
-      toggleBookmark={onClickBookmark}
-    />
+    <>
+      <View
+        {...props}
+        modal={uiControls.modal}
+        recentlySaved={uiControls.recentlySaved}
+        onOpenModal={openModal}
+        onCloseModal={closeModal}
+        onSetRecentlySaved={setRecentlySaved}
+        onSave={handleSave}
+        onCreate={handleCreate}
+        handleResize={forceUpdateDebounced}
+        toggleBookmark={onClickBookmark}
+      />
+      <Toaster
+        message="Would you like to be notified when this question is done loading?"
+        show={showToaster}
+        onDismiss={() => setShowToaster(false)}
+        onConfirm={() => handleToastConfirm()}
+        fixed
+      />
+    </>
```

Comment:
1. Should we maybe move `Toaster` to `View`?
2. Please make the `message` translatable
3. Can we avoid creating new functions for `onConfirm` and `onDismiss` on every render? It can be replaced with:

```jsx
const onDismissToast = useCallback(() => {
   setShowToaster(false);
}, [])

<Toaster
   onConfirm={handleToastConfirm}
   onDisimiss={onDismissToast}
/>
```